idaes/core/initialization/initializer_base.py

```
@@ -0,0 +1,399 @@
+class InitializerBase:
+    """
+    Base class for Initializer objects.
+
+    This implements a default workflow nad methods for common tasks.
+    Developers should feel free to overload these as necessary.
+    """
+
+    CONFIG = ConfigBlock()
+
+    CONFIG.declare(
+        "constraint_tolerance",
+        ConfigValue(
+            default=1e-5,
+            domain=float,
+            description="Tolerance for checking constraint convergence",
+        ),
+    )
+    CONFIG.declare(
+        "output_level",
+        ConfigValue(
+            default=idaeslog.NOTSET,
+            description="Set output level for logging messages",
+        ),
+    )
+
+    def __init__(self, **kwargs):
+        self.config = self.CONFIG(kwargs)
+
+        self.initial_state = None
+        self.postcheck_summary = {}
+        self.status = None
+
+    def get_logger(self, model):
+        return idaeslog.getInitLogger(model.name, self.config.output_level)
+
+    def initialize(
+        self, model: Block, initial_guesses: dict = None, json_file: str = None
+    ):
+        """
+        Execute full initialization routine.
+
+        Args:
+            model: Pyomo model to be initialized.
+            initial_guesses: dict of initial guesses to load.
+            json_file: file name of json file to load initial guesses from as str.
+
+        Note - can only provide one of initial_guesses or json_file.
+
+        Returns:
+            InitializationStatus Enum
+        """
+        # 1. Get current model state
+        self.get_current_state(model)
+
+        # 2. Load initial guesses
+        self.load_initial_guesses(
+            model, initial_guesses=initial_guesses, json_file=json_file
+        )
+
+        # 3. Fix states to make square
+        self.fix_initialization_states(model)
+
+        # 4. Prechecks
+        self.precheck(model)
+
+        # 5. try: Call specified initialization routine
+        try:
+            results = self.initialization_routine(model)
```

Why do you ignore exceptions here?  Or is it just that you haven't gotten to it?

---

In part hadn't got to it, but also letting the postcheck handle any errors that need to be raised.